server/handlers/meshery_application_handler.go

```diff
@@ -646,6 +646,55 @@ func (h *Handler) GetMesheryApplicationTypesHandler(
 	fmt.Fprint(rw, string(b))
 }
 
+// swagger: route GET /api/application/download/{id} ApplicationsAPI idGetApplication
+// Handle GET request for Meshery Application with the given id
+//
+// Get the application file with the given id
+// responses:
+//  200
+
+// GetMesheryApplicationFile returns the application file with the given id
+func (h *Handler) GetMesheryApplicationFile(
+	rw http.ResponseWriter,
+	r *http.Request,
+	_ *models.Preference,
+	_ *models.User,
+	provider models.Provider,
+) {
+	applicationID := mux.Vars(r)["id"]
+	resp, err := provider.GetMesheryApplication(r, applicationID)
+
+	if err != nil {
+		obj := "download"
+		h.log.Error(ErrApplicationFailure(err, obj))
+		http.Error(rw, ErrApplicationFailure(err, obj).Error(), http.StatusNotFound)
+		return
+	}
+
+	var mimeType string
+	switch filepath.Ext(applicationID) {
+	case ".json":
+		mimeType = "text/plain"
```

Comment:
its not required. The file is always yaml in our case.

Reply:
ok, updated